ostd/src/arch/loongarch/trap/trap.rs

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use core::arch::{asm, global_asm};
+
+use crate::arch::cpu::context::GeneralRegs;
+
+#[cfg(target_arch = "loongarch64")]
+global_asm!(
+    r"
+    .equ XLENB, 8
+    .macro LOAD_SP a1, a2
+        ld.d \a1, $sp, \a2*XLENB
+    .endm
+    .macro STORE_SP a1, a2
+        st.d \a1, $sp, \a2*XLENB
+    .endm
+"
+);
```

Comment:
What's the purpose of writing the macro definitions here? Why not just move them to `trap.S`?

Reply:
Thanks for your explanation.

The RISC-V code is copied from the [`trapframe`](https://docs.rs/trapframe/latest/trapframe/) crate, so it has support for both 32-bit and 64-bit architectures.

Asterinas does not plan to support any 32-bit architectures in the near future, so I agree that we should move this to assembly code for clarity.